app/src/server/api/routers/panel.ts

```diff
@@ -0,0 +1,63 @@
+import { createTRPCRouter, protectedProcedure } from "@/server/api/trpc";
+import { baseServerResponse, errorResponse } from "@/server/api/trpc";
+import { eq } from "drizzle-orm";
+import { reportLog, userData } from "@/drizzle/schema";
+import { fetchUser } from "@/routers/profile";
+import { getServerPusher, updateUserOnMap } from "@/libs/pusher";
+import type { UserStatus } from "@/drizzle/constants";
+import { z } from "zod";
+import { inferRouterOutputs } from "@trpc/server";
+import { nanoid } from "nanoid";
+
+export const panelRouter = createTRPCRouter({
+  forceAwake: protectedProcedure
+    .output(baseServerResponse)
+    .input(z.object({ userId: z.string() }))
+    .mutation(async ({ ctx, input }) => {
+      // Query
+      const [user, targetUser] = await Promise.all([
+        fetchUser(ctx.drizzle, ctx.userId),
+        fetchUser(ctx.drizzle, input.userId),
+        ctx.drizzle.insert(reportLog).values({
+          id: nanoid(),
```

Comment:
I'm thinking instead of logging this into reportLog, maybe we should log it into the actionLog table. The reason being we could then add information about the user into the `changes` field - e.g. I'd love to know the userData prior to reset, the battle (or non-existence) of the battle data, etc. So that we can eventually figure out why this happens in the first place - ideally this should never happen, after all.

Reply:
I changed it to actionLog and moved it after the guards and logged previous state and battleId. If you want to log more specifically it might be easier for you to add it in.